skt/reporter.py

```diff
@@ -286,15 +304,11 @@ def update_mergedata(self):
         else:
             mergedata = self.stateconfigdata(mergedata)
 
-        if not self.cfg.get('mergelog'):
-            if self.cfg.get("cfgurl"):
-                response = requests.get(self.cfg.get("cfgurl"))
-                if response:
-                    mergedata['config'] = response.text
-            else:
-                with open("%s/.config" % self.cfg.get("workdir"),
-                          "r") as fileh:
-                    mergedata['config'] = fileh.read()
+        # Retrieve the kernel config file
+        if not self.cfg.get('mergelog') and self.cfg.get('cfgurl'):
+            mergedata['config'] = self.get_config_remote()
+        elif not self.cfg.get('mergelog'):
+            mergedata['config'] = self.get_config_local()
```

Comment:
Wouldn't this be better written like this?

    if not self.cfg.get('mergelog'):
        if self.cfg.get('cfgurl'):
            mergedata['config'] = self.get_config_remote()
        else:
            mergedata['config'] = self.get_config_local()

Reply:
I can do that. I was told not to stack the `if` statements in another review. ;)

Reply:
Hmm, surely not repeating a condition is good for maintenance? @veruu, was it a request from you? What do you think of this?

Reply:
I don't remember saying anything like that (please correct me if I did) and I agree that @spbnick 's solution is more readable and clear